smoke-tests/spec/external_dns_helper.rb

```diff
@@ -78,21 +65,28 @@ def delete_txt_record(zone_id, domain_name, namespace)
 
 # Checks if the zone is empty, then deletes
 # if not empty, it will assume it contains one A record and one TXT record created by external-dns
-def cleanup_zone(zone_id, domain, namespace, ingress_name)
-  if is_zone_empty?(zone_id)
-    # delete_zone(zone_id)
-  else
-    delete_a_record(zone_id, domain, namespace, ingress_name)
-    delete_txt_record(zone_id, domain, namespace)
+def cleanup_zone(domain, namespace, ingress_name, zone_id = nil)
+  if zone_id.nil?
+    zone_id = get_zone_by_name(domain)
+  end
+
+  records = get_zone_records(zone_id)
+
+  unless is_zone_empty?(records)
+    records.each do |record|
+      case record[:type]
+        when "A"
+          delete_a_record(zone_id, record)
+        when "TXT"
+          delete_txt_record(zone_id, record)
+      end
+    end
   end
 end
 
 # Checks if a zone is empty
 # A zone is considered empty if it only contains one SOA and one NS record
-# TODO: sleep added to avoid AWS Route53 API throttling errors. Remove once that issue is resolved.
-def is_zone_empty?(zone_id)
-  sleep 1
-  records = get_zone_records(zone_id)
+def is_zone_empty?(records)
```

Comment:
If this method is only ever called from one place in the code, I would get rid of it and just inline `records > 2` as a boolean, with the explanatory comment next to it.